cli/src/main/java/dev/starfix/Starfix.java

```diff
@@ -288,37 +290,108 @@ public void setClone_path(String clone_path) {
 
     }
 
-    public static void launch_editor(Path directory, String ide, String path, String filePath) throws IOException, InterruptedException {
+    public static abstract class IDE{
+        public abstract void launch_editor(Path directory, String ide, String path, String filePath)throws IOException, InterruptedException;  
+
+    }
+
+    public static class GenericIDE extends IDE{
+
+        IDE ide;
+
+        GenericIDE(){
+            // Default Constructor
+        }
+
+        GenericIDE(String ide){
+            setIDE(ide);
+        }
+
+        public IDE getIDE(){
+            //Return Current IDE
+            return ide;
+        }
+
+        public IDE getIDE(String ide){
```

Comment:
seems there is nothing in GenericIDE besides just this method - would that not be sufficient to just have getIDE method and it returns the specific subclass as needed ?

Reply:
I actually made a method but then came across usage  of a generic launcher. Like other IDEs(VsCode, Idea etc) this class also has its launcher. You'd notice I am using that launcher in other IDE classes as well.

Reply:
But we just need one instance; why not just set a ide variable on the command and use that directly ?

Reply:
Ok Max, I'll replace that GenericIDE portion with a getIDE method.